sdl-core/src/main/scala/io/smartdatalake/statusinfo/PortUtils.scala

```diff
@@ -0,0 +1,85 @@
+package io.smartdatalake.statusinfo
+
+import io.netty.channel.unix.Errors.NativeIoException
+import io.smartdatalake.util.misc.SmartDataLakeLogger
+import org.eclipse.jetty.util.MultiException
+
+import java.net.BindException
+import scala.jdk.CollectionConverters.collectionAsScalaIterableConverter
+
+object PortUtils extends SmartDataLakeLogger {
+
+  /**
+   * Returns the user port to try when trying to bind a service. Handles wrapping and skipping
+   * privileged ports.
+   */
+  def userPort(base: Int, offset: Int): Int = {
+    (base + offset - 1024) % (65536 - 1024) + 1024
```

Comment:
What is the modulo operator doing here exactly?

Reply:
It ensure rolling over privileged ports > 65535. I added some doc and a test to clarify it. Method was copy-pasted from Spark Code at org.apache.spark.util.Utils#userPort